### PR TITLE
WIP build/tutorial changes

### PR DESCRIPTION
**Pull request by bcjordan**

### CODE REVIEW

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -296,7 +307,7 @@ APPS.forEach(function (app) {
 // Use command-line tools to run browserify (faster/more stable this way)
 var browserifyExec = 'mkdir -p build/browserified && `npm bin`/browserifyinc ' +
   '--cachefile ' + outputDir + 'browserifyinc-cache.json ' +
-  '-t reactify --extension=.jsx ' + allFilesSrc.join(' ') +
+  ' -t [ babelify --compact=false --sourceMap --sourceMapRelative="$PWD" ] -d ' + allFilesSrc.join(' ') +
```
Comment (Bjvanminnen) [inline apps/Gruntfile.js:310]: oooh. does this mean we can start using ES6 everywhere? :)

File: apps/src/appMain.js [hunk 2]
```diff
@@ -52,7 +52,8 @@ module.exports = function(app, levels, options) {
   if (studioApp.isUsingBlockly()) {
     var blockInstallOptions = {
       skin: options.skin,
-      isK1: options.level && options.level.isK1
+      isK1: options.level && options.level.isK1,
+      level: options.level
```
Comment (Bjvanminnen) [inline apps/src/appMain.js:56]: Passing the whole level object to our block installer seems a little overkill. Maybe we should just extract the properties we need and pass those along (like we do for isK1). (Note: this feedback doesnt need to block merge)

File: apps/src/canvg/rgbcolor.js [hunk 3]
```diff
@@ -1,288 +0,0 @@
...
-                xml.appendChild(list_item);
-
-            } catch(e){}
-        }
-        return xml;
-
-    }
-
-}
-
```
Comment (Bjvanminnen) [inline apps/src/canvg/rgbcolor.js:288]: Intentionally removing this file?

File: apps/Gruntfile.js (hunk 1 above)
Comment (bcjordan) [inline apps/Gruntfile.js:310]: Yeah!!

File: apps/src/appMain.js (hunk 2 above)
Comment (bcjordan) [inline apps/src/appMain.js:56]: Good point. The options in question are pretty app-specific (`.availableBlocks`, `.ifPlaceOptions`, `.placeBlockOptions`), so may want to put those in to a separate `options.level.blockOptions` object and pass that through. (Will need to work out where in the initialization process the app can do the .level.\* -> .level.blockOptions.\* translation before appMain gets called, though...)

File: apps/src/feedback.js [hunk 4]
```diff
@@ -532,7 +538,9 @@ FeedbackUtils.prototype.getFeedbackMessage_ = function(options) {
         message = msg.errorQuestionMarksInNumberField();
         break;
       case TestResults.TOO_MANY_BLOCKS_FAIL:
-        message = msg.numBlocksNeeded({
+        var messageFunction = (options.appStrings && options.appStrings.tooManyBlocksFailMsgFunction) ||
```
Comment (Bjvanminnen) [inline apps/src/feedback.js:543]: This has enough complexity that it could use a comment.

File: apps/src/turtle/turtle.js [hunk 5]
```diff
@@ -482,7 +482,9 @@ Artist.prototype.drawTurtle = function() {
       sourceY + sourceHeight > this.avatarImage.height)
   {
     if (console && console.log) {
-      console.log("drawImage is out of source bounds!");
+      // TODO(bjordan): ask Brent, starting to flood grunt mochaTest messages,
+      // better fix here?
+      // console.log("drawImage is out of source bounds!");
```
Comment (Bjvanminnen) [inline apps/src/turtle/turtle.js:487]: Why would this have changed from your stuff?

File: apps/src/canvg/rgbcolor.js (hunk 3 above)
Comment (bcjordan) [inline apps/src/canvg/rgbcolor.js:288]: Correct, rgbcolor, stackblur and canvg now get required via npm. The babelify build complained about the libraries due to some strict-mode-angering errors (global declarations, references to libraries)

**Comment by Bjvanminnen** [comment]
common.scss
the visualization changes are a little scary, just because there's already a fair amount of complexity there. we should make sure that we test on applab in particular (where we do some other funky things to the visualization). i would hope that eyes tests would catch breaks here, but im not sure they would catch bugs in things like the window being resized.

File: apps/src/turtle/turtle.js (hunk 5 above)
Comment (bcjordan) [inline apps/src/turtle/turtle.js:487]: I'm not positive, but one theory is that the change I made here to runLevelTest's `baseUrl` to support asset serving ([here](https://github.com/code-dot-org/code-dot-org/blob/2015-mc-hoc/apps/test/util/runLevelTest.js#L122-122)) might have caused the avatar image to actually get served?

baseUrl in that file used to be set to `baseUrl: '' // doesn't matter`

**Comment by bcjordan** [comment]
@Bjvanminnen that's right, was going to do a pass on confirming they are functionally equivalent. It's possible the values were tweaked since I pulled things out into that mixin. I'll confirm now.

**Comment by Bjvanminnen** [comment]
runTests.js why do we need the addition of 'web-security': false?

**Comment by Bjvanminnen** [comment]
levels_controller.rb we require json, but dont use it anywhere?

**Comment by Bjvanminnen** [comment]
_blockly.html.haml - you have a TODO(remove). not sure if that needs to be removed before merge or no

**Comment by bcjordan** [comment]
re: _blockly.html.haml, good catch, removing

**Comment by bcjordan** [comment]
re: levels_controller.rb, removing, must have been left from development

**Comment by Bjvanminnen** [comment]
initApp.js changes dont seem like they should result in any change in behavior. Should we just not make the code change and reduce churn?

**Comment by Bjvanminnen** [comment]
Things look generally okay to me. I didn't spend much time looking through stuff under src/craft as I'm assuming it won't regress other apps. It does look like there's some opportunity for a style pass (which I'm happy to participate in), but I believe that is out of scope for now. Look forward to seeing all this stuff go live :)

**Comment by bcjordan** [comment]
re: `runTests.js`, IIRC that was required to allow tests within phantom to request assets from the `:8001` asset server (because mochify runs the server by default on a different port? & doesn't allow cross-domain requests by default)

**Comment by bcjordan** [comment]
re: `initApp.js`, that change is actually functional. Previously it would also skip the video popup when `skipInstructionsPopup` was set.

Looking at usages of skipInstructionsPopup carefully now, though, it looks like that is the desired behavior of `skipInstructionsPopup` in certain cases (it gets used to not show videos/instructions on level builder and share pages).

For this app's purposes I'll introduce a different parameter for that use & restore old behavior, `skipOnlyInstructionsPopup`.

Thanks for pointing out!

File: apps/src/feedback.js (hunk 4 above)
Comment (bcjordan) [inline apps/src/feedback.js:543]: Added

**Comment by bcjordan** [comment]
Finished reviewing `application.scss` visualization mixin extraction, looks to be functionally equivalent. Will test viz resizing again manually.

**Pull request merged**